Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.cpp

```diff
@@ -89,9 +89,19 @@ WebFullScreenManager::WebFullScreenManager(WebPage* page)
 WebFullScreenManager::~WebFullScreenManager()
 {
     clearElement();
+    invalidate();
 }
 
-WebCore::Element* WebFullScreenManager::element() 
+void WebFullScreenManager::invalidate()
+{
+    clearElement();
+#if ENABLE(VIDEO)
+    setMainVideoElement(nullptr);
+#endif
+    m_mainVideoElementTextRecognitionTimer.stop();
```

Comment:
Should close() call invalidate() as well?
m_mainVideoElementTextRecognitionTimer probably deserves to be stopped when close() is called.

Reply:
good spotting, can simplify the code